### PR TITLE
WIP: upgrade webpack to v4

### PR DESCRIPTION
**Pull request by PlayMa256**
Please check out Contributing Guidelines. By following this template you help us to review your code.
https://github.com/draft-js-plugins/draft-js-plugins/blob/master/CONTRIBUTING.md

## Checklist

- [ ] Fix any eslint errors that occur
- [ ] Update change-log for every plugin you touch
- [ ] Add/Update tests if you add/change new functionality
- [ ] Add/Update docs if you add/change functionality
- [X] Enable "Allow edits from maintainers" for this PR

Upgrading webpack to v4

### CODE REVIEW

**Comment by PlayMa256** [comment]
This will be a quite piece of work :P, specially because of babel, it does not recognizes that we have a babel.config.js on the root. I'll try to make it work separately first, improve later.

**Review by sibelius — APPROVED** [review]
It is looking good

**Comment by PlayMa256** [comment]
FIY for future references, i think we could have an intermediary environment (like vercel) where we deploy PR versions of the docs if we have changes, so anyone can check it without the need to rerun it!

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-8
+12.16.3
```
Comment (sibelius) [inline .nvmrc:1]: nice

File: docs/webpack.config.base.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-var path = require('path'); // eslint-disable-line no-var
-var ExtractTextPlugin = require('extract-text-webpack-plugin'); // eslint-disable-line no-var
+const path = require('path');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
```
Comment (sibelius) [inline docs/webpack.config.base.js:2]: we don't need extract-text-webpack-plugin ?

Comment (PlayMa256) [reply docs/webpack.config.base.js:2]: Nop! That is used to extract css from code into .css files. `mini-css-extract-plugin` is the new plugin for it. `extract-text-webpack-plugin` is "deprecated"

**Comment by PlayMa256** [comment]
prismjs is killing me to solve the css files. Does anyone has any idea?

**Comment by PlayMa256** [comment]
ok, it does not load now, it is something...

File: docs/babel.config.js [hunk 3]
```diff
@@ -1,7 +1,21 @@
+const removeLinariaImport = () => ({
```
Comment (sibelius) [inline docs/babel.config.js:1]: nice

File: yarn-error.log [hunk 4]
```diff
@@ -0,0 +1,15551 @@
+Arguments:
```
Comment (TrySound) [inline yarn-error.log:1]: I think better to remove this

Comment (PlayMa256) [reply yarn-error.log:1]: Yep, forgot this! thanks!

**Pull request closed without merging**